tests/integration/nm/ovs_test.py

```diff
@@ -0,0 +1,185 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import pytest
+
+from libnmstate import nm
+from libnmstate.schema import OVSBridge as OB
+
+from .testlib import mainloop
+
+
+BRIDGE0 = 'brtest0'
+ETH1 = 'eth1'
+
+
+@pytest.fixture
+def bridge_minimum_config():
+    return {
+        OB.CONFIG_SUBTREE: {
+            OB.PORT_SUBTREE: []
+        }
+    }
+
+
+@pytest.fixture
+def bridge_default_config():
+    return {
+        OB.CONFIG_SUBTREE: {
+            OB.OPTIONS_SUBTREE: {
+                OB.FAIL_MODE: '',
+                OB.MCAST_SNOOPING_ENABLED: False,
+                OB.RSTP: False,
+                OB.STP: False
+            },
+            OB.PORT_SUBTREE: []
+        }
+    }
+
+
+def test_create_and_remove_minimum_config_bridge(bridge_minimum_config,
+                                                 bridge_default_config):
+    bridge_desired_state = bridge_minimum_config
+
+    with _bridge_interface(bridge_desired_state):
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_current_state == bridge_default_config
+        assert bridge_current_state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE] == []
+
+    assert not _get_bridge_current_state()
+
+
+def test_bridge_with_system_port(eth1_up, bridge_default_config):
+    bridge_desired_state = bridge_default_config
+
+    eth1_port = {
+        OB.PORT_NAME: 'eth1',
+        OB.PORT_TYPE: 'system',
+        # OVS vlan/s are not yet supported.
+        # OB.PORT_VLAN_MODE: None,
+        # OB.PORT_ACCESS_TAG: 0,
+    }
+
+    bridge_desired_state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE].append(eth1_port)
+
+    with _bridge_interface(bridge_desired_state):
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_desired_state == bridge_current_state
+
+    assert not _get_bridge_current_state()
+
+
+@contextmanager
+def _bridge_interface(state):
+    try:
+        _create_bridge(state)
+        yield
+    finally:
+        _delete_iface(BRIDGE0)
+        for p in state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE]:
+            _delete_iface(nm.ovs.PORT_PROFILE_PREFIX + p[OB.PORT_NAME])
+            _delete_iface(p[OB.PORT_NAME])
+
+
+def _get_bridge_current_state():
```

Comment:
For me the feelings that the test cases should be classes becomes stronger. Given that we have many methods that operate on the constants instead of (only) their arguments lets me think that it would be nice to have a BridgeTestcase class with properties such as `bridge_name` and methods like `_get_current_state(self)`. It is somehow like a class implemented as a module currently.

Reply:
Currently it is just forking the Linux bridge nm tests.
Refactoring this is a good idea and it should probably fit the Linux bridge one as well.

Could you please add it as an issue and assign it to the current sprint?